raiden/utils/upgrades.py

```diff
@@ -176,13 +177,28 @@ def run(self):
             log.debug(f'Upgrading database from {older_version} to v{RAIDEN_DB_VERSION}')
 
             try:
+                target_version = older_version
                 with storage.transaction():
-                    version_iteration = older_version
-                    for upgrade_func in UPGRADES_LIST:
-                        version_iteration = _run_upgrade_func(
+                    for source_version, upgrade_func in UPGRADES_LIST.items():
+                        # Source_version represents the version number
+                        # from which a migration starts to upgrade to a target version.
+                        # The target version is returned by the upgrade_func.
+                        # The below step basically skips a migration in case
+                        # the **latest older** version we have is larger than
+                        # the current migration's source version.
+                        # Example, assuming some node was running version 18 and
+                        # Upgraded to 20.
+                        # In this case, only migrations:
+                        # - 18_to_19
+                        # - 19_to_20
+                        # should run.
+                        if source_version < target_version:
```

Comment:
The variable naming here is really confusing.

I would expect target_version to always be constant and not change every iteration. What you call target_version here should be `iterated_version` or something.

This is still a counter-intuitive way to upgrade imo. You can see it for yourself too by looking at the fact that you have to write such a big comment to explain stuff here.

Why not just keep it simple? You can keep the list as it was. Just turn it into a list of `UpgradeRecords` (A tuple of version and upgrade funcs), and simply start  iterating from the point of our `upgrade_starting_version`?

I also dislike that you rely on the dict being ordered which only happens in python 3.6 and afterwards.